legend-depot-core-http/src/main/java/org/finos/legend/depot/tracing/resources/BaseResource.java

```diff
@@ -94,12 +98,37 @@ protected <T> T handle(String resourceAPIMetricName, String label, Supplier<T> s
         return TracerFactory.get().executeWithTrace(label, () -> handleWithLogging(resourceAPIMetricName, label, supplier));
     }
 
+    protected <T> Response handle(String resourceAPIMetricName, String label, Supplier<T> supplier, Request request, Supplier<EntityTag> etagSupplier)
+    {
+        EntityTag serverTag;
+        try
+        {
+            serverTag = etagSupplier.get();
```

Comment:
if the etag calculation fails... would be good to at least log some sort of error